boto/endpoints.py

```diff
@@ -134,6 +130,53 @@ def _service_name(self, endpoint_prefix):
         return self._service_name_map.get(endpoint_prefix, endpoint_prefix)
 
 
+class BotoEndpointResolver(object):
+    """Resolves endpoint hostnames for AWS services.
+
+    This is NOT intended for external use.
```

Comment:
I think this is appropriate wording. While you can use it, it is not recommended as it is not intended.